app/controllers/songs_controller.rb

```diff
@@ -1,5 +1,34 @@
 class SongsController < ApplicationController
   def index
+    respond_to do |format|
+      format.json do
+        songs = Song.all
+        total_songs = songs.count
+
+        if params[:sSearch].present?
+          search = {
+            query: "%#{params[:sSearch]}%",
+            rQuery: Regexp.escape(params[:sSearch])
+          }
+
+          songs = songs.where('name LIKE :query OR artist LIKE :query OR
```

Comment:
The idea is that since it's the index page you want to load all of the songs (paginated) so that the user can browse through them. If they choose to filter then we narrow down the query. That's why the `songs.where` is nested in the block which checks if the search is present. 

As for abstracting the query logic, sounds good to me. I don't know how you want it done so I'll let you handle that.